### PR TITLE
[ MB-13764, MB-15006, MB-14736 ] Updating the GEX URL in only Staging to test new GEX HTTPS connection

### PR DESCRIPTION
**Pull request by rogeruiz**
## Summary

This work was done in #9789 which was later reverted in #9796 by @reggieriser.
We're now testing this works again with GEX DLA so I'm updating the codebase to
use the new URL here.

## Setup to Run Your Code

<details>
<summary>💻 You will need to test this within an ATO environment such as Staging
or Production. These instructions assume Staging.</summary>

##### Deploy this PR to Staging by getting it into the default branch

This code needs to run behind the BCAP and will need to be deployed in an ATO
environment.

##### Approve a move for the Prime API

As a Customer user and a Office user, you will need to get a move into a state that
it can be viewed by the Prime Simulator user.

##### Submit a payment request as the Prime API

As the Prime Simulator user, you will have to submit a payment request so that
the ECS Scheduled task which runs every hour, picks up the job and submits it to
GEX.

##### View the CloudWatch logs to ensure that it's working correctly

Once it's deployed, you will have to login to the AWS UI console to view the
CloudWatch Log Insights to view the GEX ECS job to ensure that it's working.

</details>

### CODE REVIEW

File: cmd/milmove-tasks/process_edis.go [hunk 1]
```diff
@@ -167,6 +167,7 @@ func processEDIs(cmd *cobra.Command, args []string) error {
 	appCtx := appcontext.NewAppContext(dbConnection, logger, nil)
 	dbEnv := v.GetString(cli.DbEnvFlag)
 	gexURL := v.GetString(cli.GEXURLFlag)
+	logger.Info(fmt.Sprintf("GEX URL is %v", gexURL))
```
Comment (rogeruiz) [inline cmd/milmove-tasks/process_edis.go:170]: I'll be leaving this as it will be helpful when looking at GEX logs.

File: .envrc [hunk 2]
```diff
@@ -200,7 +200,7 @@ export GEX_BASIC_AUTH_USERNAME="mymovet"
 require GEX_BASIC_AUTH_PASSWORD "See 'DISABLE_AWS_VAULT_WRAPPER=1 AWS_REGION=us-gov-west-1 aws-vault exec transcom-gov-dev -- chamber read app-devlocal gex_basic_auth_password'"
 export GEX_URL=""
 # To actually send the GEX request, replace url in envrc.local with the line below
-# export GEX_URL=https://gexweba.daas.dla.mil/msg_data/submit/
+# export GEX_URL=https://gexb.gw.daas.dla.mil/msg_data/submit/
```
Comment (rogeruiz) [inline .envrc:203]: These changes are all similar enough where it's just the subdomains being modified. This new URL was contained in the latest emails from J Mc from GEX DLA.

File: pkg/cli/gex.go [hunk 3]
```diff
@@ -34,6 +34,7 @@ const (
 var gexHostnames = []string{
 	"gexweba.daas.dla.mil",
 	"gexwebb.daas.dla.mil",
```
Comment (rogeruiz) [inline pkg/cli/gex.go:36]: I believe we should eventually remove these as we won't be using these host names in the future. I can tackle this in another PR. But I wanted to call them out here for folks to see.

**Review by rogeruiz — COMMENTED** [review]
Leaving some comments to aid in the review of this PR.

File: pkg/cli/gex.go (hunk 3 above)
Comment (rogeruiz) [reply pkg/cli/gex.go:36]: The way they're currently getting used is to validate that the `GEX_URL` being passed in matches the hard-coded hostnames in the Golang code.

**Review by reggieriser — COMMENTED** [review]
This LGTM, but in comparing to my previous PR, we may have missed one config file to change here?  https://github.com/transcom/mymove/blob/8fb69f823c2dae3527d43905290321d6367c9969/config/env/exp.app-client-tls.env#L13

**Review by reggieriser — APPROVED** [review]
LGTM, although for some reason the `client_test` is failing due to decreased test coverage, although you're not changing any front-end code.

**Review by ahobson — APPROVED** [review]
🚀

**Comment by rogeruiz** [comment]
Rerunning the `ci/circleci:client_test` from failed to see if it can pass, but I will modify the CircleCI cache to reset it if I get stuck again.

**Comment by rogeruiz** [comment]
Moving this into Draft so it doesn't get merged in. This will eventually be merged into `main` though once the new URL is tested and in place.

**Comment by ahobson** [comment]
> Rerunning the `ci/circleci:client_test` from failed to see if it can pass, but I will modify the CircleCI cache to reset it if I get stuck again.

Any chance you need to merge in the lastest from main?

**Comment by rogeruiz** [comment]
@mergifyio rebase

**Comment by rogeruiz** [comment]
Moving this back into DRAFT so it doesn't get merged in. I will be able to pick this back up after 13:00 ET today.

**Comment by rogeruiz** [comment]
GEX has confirmed that they have deactivated the old GEX endpoints so this PR will now be used to replace everything in the application code for the new GEX endpoints.

**Comment by rogeruiz** [comment]
Failing on client_test code coverage again. I'll troubleshoot this in a Huddle / Hangout on Monday. 👋🏽

**Pull request merged**